### PR TITLE
feat: Add workflow to update tap based on a tag

### PR DESCRIPTION
**Pull request by phated**
Same logic as the website PR, we'll be using the GitHub API to trigger this from the main grain repo.

I also removed the workflows that homebrew created when we scaffolded this repo.

### CODE REVIEW

**Comment by phated** [comment]
Example of this in action: https://github.com/phated/homebrew-tap/commit/0b08bdb68742f530ec704b4fe0e192ee28a901d6

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          ln -s $(pwd) ${{ env.GRAIN_TAP_PATH }}
+
+      - name: Configure git
+        run: |
+          git config user.name grainbot[bot]
+          git config user.email bot@grain-lang.org
+
+      - name: Update tap
+        run: |
+          brew bump-cask-pr --version ${{ env.GRAIN_VERSION }} --write --commit grain
```
Comment (ospencer) [inline .github/workflows/release.yml:43]: Ohhhhhhhh, so this is how you do it 😛

**Review by ospencer — APPROVED** [review]
Awesome!

File: .github/workflows/release.yml (hunk 1 above)
Comment (phated) [reply .github/workflows/release.yml:43]: It's buried **deeeeeeeep** in their docs/manpage but I eventually found it. They mostly use it to update the Homebrew repo directly, but I figured out how we can ~~abuse~~ use it 😉

**Pull request merged**